### PR TITLE
Blog - How to Build Java Projects with Bazel

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@balapriyac {preview link](https://6465fc9d22467a147fdc7570--ephemeral-narwhal-3ac3b2.netlify.app/blog/build-java-projects-with-bazel)

File: blog/_posts/2023-06-22-build-java-projects-with-bazel.md [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+    name = "Salutations",
+    main_class = "com.bazel.example.Salutations",
+    srcs = ["src/main/java/com/bazel/example/Salutations.java"],  
+    deps = ["@maven//:org_beryx_text_io"], 
+)
+~~~
+
+This adds the `org.beryx.text-io` dependency to your project. The `deps` rule uses the [Canonical](https://developers.google.com/search/docs/crawling-indexing/canonicalization) path to point toward the dependencies. Any slashes and dashes in the package name are replaced with underscores (`_`).
+
+> That's how you add external dependencies, but what about local dependencies? The best way to add a local dependency is to split your project and create separate targets for your local libraries or class files.
```
Comment (balapriyac) [inline blog/_posts/2023-06-22-build-java-projects-with-bazel.md:199]:
```suggestion
<div class="notice--big--primary">
**That's how you add external dependencies, but what about local dependencies?** The best way to add a local dependency is to split your project and create separate targets for your local libraries or class files.
</div>
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer.

**Comment by msali123** [comment]
@balapriyac  [Changes Done](https://646b1f4e4981342b34d83f8f--ephemeral-narwhal-3ac3b2.netlify.ap)

**Review by jalletto — APPROVED** [review]

**Pull request merged**